spring-cloud-netflix-ribbon/src/main/java/org/springframework/cloud/netflix/ribbon/PropertiesFactory.java

```diff
@@ -39,6 +39,9 @@ public String getClassName(Class clazz, String name) {
 		if (this.classToProperty.containsKey(clazz)) {
 			String classNameProperty = this.classToProperty.get(clazz);
 			String className = environment.getProperty(name + "." + NAMESPACE + "." + classNameProperty);
+			if(!StringUtils.hasText(className)){
```

Comment:
Should we return `null` in the case where it is the empty string?

Reply:
when empty, it should return environment.getProperty(NAMESPACE + "." + classNameProperty);
this is for setting default properties: 
in https://github.com/Netflix/ribbon/wiki/Programmers-Guide , descript:
  If a property is missing the clientName, it is interpreted as a property that applies to all clients.